lib/model/fakeconns_test.go

```diff
@@ -33,7 +33,7 @@ type fakeConnection struct {
 	folder                   string
 	model                    *model
 	indexFn                  func(string, []protocol.FileInfo)
-	requestFn                func(folder, name string, offset int64, size int, hash []byte, fromTemporary bool) ([]byte, error)
+	requestFn                func(folder, name string, offset int64, size int, hash []byte, fromTemporary bool, ctx context.Context) ([]byte, error)
```

Comment:
convention is to have this as the first argument.

Reply:
also, this could reuse protocol.BufferPool 🙃

Reply:
Don't get it.

Also should have addressed all the comments.